src/Symfony/Bundle/FrameworkBundle/Templating/TemplateResponse.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Templating;
+
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @author Pierre du Plessis <pdples@gmail.com>
+ */
+class TemplateResponse implements TemplateResponseInterface
```

Comment:
Thinking about this again, I don't think the interface should be related to a template response at all. The interface defines a way to return a Response. The `Template(d)Response` class is just an implementation rendering a template, but you can just as well return a `JsonResponse`, which means it's no longer rendering a template. So the name should probably be something more generic, although I'm not sure if just `ResponseInterface` will do?